scanner/templates/node/Dockerfile

```diff
@@ -1,38 +1,68 @@
-FROM debian:bullseye as builder
+# syntax = docker/dockerfile:1
 
-ENV PATH=/usr/local/node/bin:$PATH
+# Adjust NODE_VERSION as desired
 ARG NODE_VERSION={{ .nodeVersion }}
+FROM node:${NODE_VERSION}-slim as base
+
+LABEL fly_launch_runtime="{{ .runtime }}"
+
+# {{ .runtime }} app lives here
+WORKDIR /app
+
+# Set production environment
+ENV NODE_ENV=production
 {{ if .yarn -}}
+
 ARG YARN_VERSION={{ .yarnVersion }}
+RUN npm install -g yarn@$YARN_VERSION
 {{ end }}
-RUN apt-get update; apt install -y curl python-is-python3 pkg-config build-essential && \
-    curl -sL https://github.com/nodenv/node-build/archive/master.tar.gz | tar xz -C /tmp/ && \
-    /tmp/node-build-master/bin/node-build "${NODE_VERSION}" /usr/local/node && \
-{{ if .yarn -}}
-    npm install -g yarn@$YARN_VERSION && \
-{{ end -}}
-    rm -rf /tmp/node-build-master
 
-RUN mkdir /app
-WORKDIR /app
+# Throw-away build stage to reduce size of final image
+FROM base as build
+
+# Install packages needed to build node modules
+RUN apt-get update -qq && \
+    apt-get install -y python-is-python3 pkg-config build-essential {{
+    if .prisma }}openssl {{ end }}
+
+# Install node modules
+COPY --link {{ .package_files }} .
+RUN {{ .packager }} install{{ if .devDependencies }} --production=false{{ end }}
 
-COPY . .
+{{ if .prisma -}}
+# Generate Prisma Client
+COPY --link prisma .
+RUN npx prisma generate
+
+{{ end -}}
+# Copy application code
+COPY --link . .
 
 {{ if .build -}}
-RUN {{ .packager }} install && {{ .packager }} run build
+# Build application
+RUN {{ .packager }} run build
+
+{{ end -}}
+{{ if .devDependencies -}}
+# Remove development dependencies
+{{ if .yarn -}}
+RUN yarn install --production=true
 {{ else -}}
-RUN {{ .packager }} install
+RUN npm prune --production
+{{ end -}}
+
 {{ end }}
 
-FROM debian:bullseye-slim
+# Final stage for app image
+FROM base
 
-LABEL fly_launch_runtime="nodejs"
+# Copy built application
+COPY --from=build /app /app
 
-COPY --from=builder /usr/local/node /usr/local/node
-COPY --from=builder /app /app
-
-WORKDIR /app
-ENV NODE_ENV production
-ENV PATH /usr/local/node/bin:$PATH
+{{ if .prisma -}}
+# Entrypoint prepares the database.
+ENTRYPOINT ["/app/docker-entrypoint"]
 
+{{ end -}}
+# Start the server by default, this can be overwritten at runtime
 CMD [ "{{ .packager }}", "run", "start" ]
```

Comment:
Can we use the same way of running the app?

We updated the dockerfile to _always_ use the script file, so that both production & local development are running the exact same thing in https://github.com/remix-run/indie-stack/pull/98